src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,296 @@
+    It checks if all the columns from the set are in the pd.DataFrame.
+    Additional columns to the ones in the set are allowed.
+    It raises an error if any columns are missing.
+
+    Args:
+        df: pd.DataFrame that is checked
+        required_columns: set of column names
+
+    Raises:
+        UserError: if there are required columns missing
+    """
+    if not required_columns.issubset(set(df.columns)):
+        raise UserError(
+            f"The following columns are missing in the excel:\n" f"{required_columns.difference(set(df.columns))}"
+        )
+
+
+def check_column_for_duplicate_else_raise_error(df: pd.DataFrame, to_check_column: str) -> None:
+    """
+    This function checks if a specified column contains duplicate values.
+    Empty cells (pd.NA) also count as duplicates.
+    If there are any duplicate values, it creates a string with the duplicates which are displayed in the error message.
+
+    Args:
+        df: pd.DataFrame that is checked for duplicates
+        to_check_column: Name of the column that must not contain duplicates
+
+    Raises:
+        UserError: if there are duplicates in the column
+    """
+    if df[to_check_column].duplicated().any():
+        # If it does, it creates a string with all the duplicate values and raises an error.
+        duplicate_values = ",".join(df[to_check_column][df[to_check_column].duplicated()].tolist())
+        raise UserError(
+            f"The column '{to_check_column}' may not contain any duplicate values. "
+            f"The following values appeared multiple times '{duplicate_values}'."
+        )
+
+
+def check_required_values(df: pd.DataFrame, required_values_columns: list[str]) -> dict[str, pd.Series]:
+    """
+    If there are any empty cells in the column, it adds the column name and a boolean pd.Series to the dictionary.
+    If there are no empty cells, then it is not included in the dictionary.
+    If no column has any empty cells, then it returns an empty dictionary.
+
+    Args:
+        df: pd.DataFrame that is checked
+        required_values_columns: a list of column names that may not contain empty cells
+
+    Returns:
+        a dictionary with the column names as key and pd.Series as values if there are any empty cells
+    """
+    # It checks if any of the values in a specified column are empty. If they are, they are added to the dictionary
+    # with the column name as key and a boolean series as value that contain true for every pd.NA
+    res_dict = {col: df[col].isnull() for col in required_values_columns if df[col].isnull().any()}
+    # If all the columns are filled, then it returns an empty dictionary.
+    return res_dict
+
+
+def turn_bool_array_into_index_numbers(series: pd.Series[bool], true_remains: bool = True) -> list[int]:
+    """
+    This function takes a pd.Series which only contains boolean values, by default, the index numbers of the True values
+    are extracted.
```

Comment:
I don't understand the first sentence of the docstring. I think it's grammatically not correct.